test/function_test.py

```diff
@@ -496,6 +496,24 @@ def test_function_exception(client, servicer):
         assert "foo!" in str(excinfo.value)
 
 
+def test_function_exception_gather(client, servicer):
+    app = App()
+
+    servicer.function_body(failure)
+    failure_modal = app.function()(failure)
+    with app.run(client=client):
+        with pytest.raises(CustomException) as excinfo:
+            FunctionCall.gather(failure_modal.spawn(), failure_modal.spawn())
+        assert "foo!" in str(excinfo.value)
+
+    with app.run(client=client):
+        results = FunctionCall.gather(failure_modal.spawn(), failure_modal.spawn(), return_exceptions=True)
+        for result in results:
+            assert isinstance(result, UserCodeException)
```

Comment:
What is this `UserCodeException` type?

Reply:
After some more investigation, the problem might be worse than I thought, apparently the client has some explicit wrapping using UserCodeException too in _process_result, which seems like the more likely culprit here 🤯

Reply:
We need to get a 😵‍💫 option for GitHub reactions 😁

Reply:
Ohhhh no, `synchronicity.UserCodeException` is apparently being returned by `.map(return_exceptions=True)` as well :(

Now I'm not sure what to do - will have to discuss in client team...